src/h5web/visualizations/heatmap/PixelAxis.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { Axis } from '@vx/axis';
+import { scaleLinear } from 'd3-scale';
+import { format } from 'd3-format';
+import { AxisOrientation } from '@vx/axis/lib/types';
+import styles from './HeatmapVis.module.css';
+
+interface Props {
+  orientation: AxisOrientation;
+  numberPixels: number;
+  width: number;
+  height: number;
+}
+
+const axisStyles: Record<AxisOrientation, string> = {
+  left: styles.axisLeft,
+  bottom: styles.axisBottom,
+  top: '',
+  right: '',
+};
+
+function PixelAxis(props: Props): JSX.Element {
+  const { orientation, numberPixels, width, height } = props;
+  const range = ['top', 'bottom'].includes(orientation)
+    ? [0, width]
+    : [0, height];
+
+  const scale = scaleLinear()
+    .domain([-0.5, numberPixels - 0.5]) // Shift domain so that integer ticks are in the middle of the pixels
+    .range(range);
+
+  const axisProps = {
+    left: 0,
+    orientation,
+    scale,
+    strokeWidth: 4,
+    tickFormat: format('.0'),
+    numTicks: Math.min(numberPixels, 5),
+  };
+  const tickLabelProps = {
+    className: styles.tick,
+    dx: '-0.25em',
+    dy: '0.25em',
+  };
+  // Adds extra X shifting for AxisLeft
+  if (orientation === 'left') {
+    axisProps.left = width;
+    tickLabelProps.dx = '-0.75em';
+  }
+
+  return (
+    <svg className={axisStyles[orientation]} style={{ width, height }}>
```

Comment:
You could put `orientation` in a data attribute to avoid having to keep a record of styles.